Golden repo merge

techprojsforfun opened the pull request:
merge all code from iterations into golden repo!

Commit:
Added rspec tests and route/controller action/view for favorited recipes

Commit:
refactor find_in_api method, add helpers

Commit:
Merge pull request #29 from omidzargham/refactor_recipes_model

refactor find_in_api method, add helpers

Commit:
refactor show method

ChloeGuyue committed:
added view

ChloeGuyue committed:
merge

Commit:
more show refactoring

Commit:
remove byebug

ChloeGuyue committed:
fixed bug

ChloeGuyue committed:
more view

ChloeGuyue committed:
fix bug

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into favorites_model

Commit:
Merge pull request #30 from omidzargham/favorites_model

Added ability to favorite recipes and view favorites by meal type (w/ Muxuan)

Commit:
new api testing

Commit:
update with master

Commit:
Merge pull request #31 from omidzargham/refactor_show_method

Refactor show method

Commit:
change api and refactor accordingly

Commit:
merge with master on add_filter_readyinminutes

Commit:
fix merge conflict residue

Commit:
re-add throttle sleeper

Commit:
force travis rebuild

Commit:
rebuild travis build AGAIN

Commit:
rebuild travis build yet AGAIN

Commit:
another travis rebuild

Commit:
Merge pull request #32 from omidzargham/add_filter_readyinminutes

Add filter readyinminutes - WITH ANTHONY AND CJ

Commit:
remove sleep throttling:

Commit:
Merge pull request #33 from omidzargham/remove_sleep_throttle

remove sleep throttling:

Commit:
restructured stylesheet structure

nkompella committed:
Merge pull request #34 from omidzargham/restructure-stylesheets

restructured stylesheet structure

Commit:
Created groceries model

Commit:
Favoriting a recipe now saves the grocery list for that recipe in db

ChloeGuyue committed:
added groceries view

Commit:
Renamed grocery to grocery_list and made some cosmetic changes

Commit:
added navbar partial

Commit:
remove old links

Commit:
added navbar styling

Commit:
limit navbar options when user is signing up

Commit:
formatted alerts

Commit:
switched back to real data

ahz0001 committed:
progress on cuisineType.  seems might need to put into query

cheolhojang committed:
REMOVED ANYTHING ABOUT PRICE

Commit:
created side-navbar with slide in slide out

Commit:
reformatted alerts

Commit:
linked alerts scss to application.scss

Commit:
removed alerts from obstructing views

cheolhojang committed:
Merge pull request #36 from omidzargham/no_prices

REMOVED ANYTHING ABOUT PRICE

nkompella committed:
Merge branch 'master' into add-navbar

Commit:
fixed merge issue

Commit:
removed duplicate lines in show.html.haml

Commit:
Converted factory girl to factory bot gem and added user factory in cucumber

cheolhojang committed:
link to recipe

nkompella committed:
Merge pull request #37 from omidzargham/add-navbar

Add navbar

cheolhojang committed:
Merge branch 'master' into add_recipe_link

Commit:
API stubbed, all tests passing

Commit:
added topbar to all pages

Commit:
changed back to real api call

Commit:
repushing to rebuild

Commit:
my commits from topbar pr

nkompella committed:
Merge pull request #40 from omidzargham/add-topbar

added topbar to all pages

Commit:
started styling signup page

Commit:
integrated image into signup view

Commit:
fixed horizontal scroll issue

cheolhojang committed:
Merge pull request #38 from omidzargham/add_recipe_link

link to recipe

Commit:
centered, aligned and added box to sign up module

Commit:
changed font to helvetica

nkompella committed:
Merge branch 'master' into ui-signup-view

Commit:
Merge pull request #39 from omidzargham/stub_cucumber_tests

Stub cucumber tests

ahz0001 committed:
Merge branch 'master' of github.com:omidzargham/Grocer-Plus into add_cuisine_type

ahz0001 committed:
Switched over from cuisine preferences to dietary preferences due to API error

Commit:
added body class to signup-view to make css more specific

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into grocery_list

Commit:
imported new Fonts and fixed spacing

Commit:
moved login button below module

Commit:
differentiated login class labels

Commit:
Added rspec test for committing meal plan recipes to db

Commit:
rename all cuisine to diet, including in db

Commit:
bootstrap is my lord and savior

Commit:
changed hover button color

Commit:
Added flow for when to call API/when to commit recipes to db

Commit:
Updated check to see if API should be called in show method

Commit:
removed unnecessary testing line

Commit:
Implemented convert_to_recipe method in users controller

Commit:
commented out fake data

nkompella committed:
Merge pull request #41 from omidzargham/ui-signup-view

Updated UI for Signup View

Commit:
gave login page the same styling as signup page

Commit:
removed navbar in welcome pages, fixed expansion issue in internal pages

Commit:
Added model for MealPlanRecipe

Commit:
added welcome bar to all devise pages

Commit:
added to initial edit preferences form as well

Commit:
apis like this make me hate coding

Commit:
Changed relationship of users to recipes to be one-to-many

nkompella committed:
Merge pull request #43 from omidzargham/fix-navbar-issues

Fixed Navbar issues

Commit:
formatted form labels and centered

Commit:
Passing test to commit recipes to db

nkompella committed:
Merge branch 'master' into ui-login-view

Commit:
fixed merge conflicts with master

Commit:
add migration file for renaming cuisine to dietary_preferences

nkompella committed:
Merge pull request #42 from omidzargham/ui-login-view

Updated UI for Login View

Commit:
fixed error on signup page

Commit:
merged master into branch

Commit:
Added rspec test to ensure api called when ':api' in params

Commit:
fixed form input spacing

Commit:
added image to edit_page

Commit:
finished edit page with existing vars

Commit:
updated cucumber tests to match new labeling

Commit:
integrated bootstrap cards with flexbox support

Commit:
removed extraneous flexboxing

Commit:
Meal plan recipes committed, API only called upon form submission

Commit:
Added calories attribute to users so calories are remembered when show page is refreshed

Commit:
fixed favoriting recipes functionality

Commit:
fixed recipe links

Commit:
styling with mock images

Commit:
finished page layout

Commit:
added show_img asset

Commit:
fixed photo rendering issue

Commit:
finished yayy

Commit:
added background img

Commit:
reverted stupid api call

Commit:
increased font thickness

Commit:
added welcome message and styled background image

Commit:
added sign up, log in buttons and stylized content

Commit:
Added image and link to factory

Commit:
Merge pull request #45 from omidzargham/commit_recipes_db

Meal Plan recipes are persisted, API only called when form submitted

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into grocery_list

ChloeGuyue committed:
grocery_list view

Commit:
Grocery list generated from meal plan recipes, instead of favorited recipes

Commit:
Merge branch 'grocery_list' of https://github.com/omidzargham/Grocer-Plus into grocery_list

Commit:
Passing rspec tests, added test for rendering grocery list view

Commit:
Passing cucumber tests

Commit:
Merge branch 'master' into add_cuisine_type

Commit:
apply migrations

nkompella committed:
Merge pull request #47 from omidzargham/ui-mealplan-view

Updated UI for Mealplan View

Commit:
Merge pull request #44 from omidzargham/add_cuisine_type

Add cuisine type - PAIR WITH ANTHONY

Commit:
merged master into branch, cleaned up extraneous files

nkompella committed:
Merge pull request #46 from omidzargham/ui-edit-view

Updated UI for Edit View

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into grocery_list

Commit:
merged master into branch

Commit:
added log out page

Commit:
Fixed button spacing

Commit:
brought down button size

Commit:
changed font styles on logout page

Commit:
Rerun migration

Commit:
Added check in favorite_recipe action to ensure recipe not added twice

Commit:
formatted alerts

Commit:
removed jquery

Commit:
fix for dietary preferences checkboxes not working properly

Commit:
Updated cucumber tests for favoriting recipes

Commit:
Merge pull request #51 from omidzargham/fix_dietary_prefs

fix for dietary preferences checkboxes not working properly

Commit:
cleaned up forgot our password page

nkompella committed:
Merge pull request #49 from omidzargham/ui-home-view

Updated UI for Home and Logout Views

nkompella committed:
Merge branch 'master' into ui-fix-alerts

nkompella committed:
Merge pull request #50 from omidzargham/ui-fix-alerts

Updated UI for Alerts/Notices

Commit:
Cleaned up more devise pages

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into prevent_duplicate_favorites

Commit:
updated edit password ui

Commit:
updated resend unlock instructions page

Commit:
updated confirmation instructions ui

Commit:
updated email_changed ui

Commit:
updated password change ui

Commit:
updated reset password instructions

Commit:
updated all elements to fall within the dv-form formatting scope

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into ui-devise-cleanup

Commit:
this api makes me want to kill myself

Commit:
Fixed issue where favoriting a recipe and calling API again caused error

Commit:
Merge pull request #52 from omidzargham/prevent_duplicate_favorites

Prevent duplicate favorites

Commit:
Merge pull request #53 from omidzargham/ui-devise-cleanup

UI Devise Cleanup

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into grocery_list

Commit:
Added titles to test data, fixed issue where favorited recipes mistakenly showing as plan recipes

Commit:
Cucumber scenario for grocery list

Commit:
Rspec tests for grocery list updated

Commit:
Merge pull request #48 from omidzargham/grocery_list

Added Grocery List (with Muxuan)

Commit:
Fixed error thrown when submitting email for forgot password screen. No email sending though

ChloeGuyue committed:
added api alert

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into api_alert

Commit:
Redirect to edit page, but flash message not showing

Commit:
Renamed view variable

Commit:
added 3 grocery cards

Commit:
created flexbox grocery list view

Commit:
Removed syntax error

Commit:
finished grocery view

nkompella committed:
Merge pull request #55 from omidzargham/ui-grocery-view

Updated UI for Grocery View

Commit:
converted favorites view to erb

Commit:
Removed html

Commit:
added styling for favorites page

Commit:
added image

cheolhojang committed:
round 1 of uncomments

cheolhojang committed:
round 2 of uncomments

cheolhojang committed:
websteps comments

cheolhojang committed:
extra

cheolhojang committed:
database comments removed

cheolhojang committed:
stuff about price

Commit:
styled flexbox

Commit:
centered labeling

nkompella committed:
Merge pull request #56 from omidzargham/ui-favorites-view

Updated UI for Groceries View

Commit:
updated navbar style

cheolhojang committed:
stuff

ChloeGuyue committed:
deleted some commits

Commit:
fixed edit page spacing issues

Commit:
Fixed spacing issues

cheolhojang committed:
refactored calc_calories

Commit:
more spacing

cheolhojang committed:
cleanup

Commit:
restructured sass files

Commit:
cleaned up views

Commit:
api alert displaying. yay

ChloeGuyue committed:
Merge pull request #57 from omidzargham/comment

deleted some comments

Commit:
Merge pull request #54 from omidzargham/api_alert

added api alert

nkompella committed:
Merge pull request #59 from omidzargham/ui-final-cleanup

UI Final Cleanup

cheolhojang committed:
merge conflicts

cheolhojang committed:
Merge branch 'master' into cleanup

cheolhojang committed:
readjust

cheolhojang committed:
Adjusted tests

cheolhojang committed:
merge

cheolhojang committed:
Merge pull request #58 from omidzargham/cleanup

Cleanup

ahz0001 committed:
Fixing tests affected by refactoring

Commit:
Improved cognitive complexity of calc_calories

Commit:
Show method refactored

Commit:
Fixed issue in set_vars_from_recipe

Commit:
Fixed minor errors

Commit:
add link to instructions for favorited recipes

Commit:
fix root routing bug

Commit:
Controller filter for logged in user

Commit:
cleanup

Commit:
Merge pull request #61 from omidzargham/add_favorite_recipe_link

add link to instructions for favorited recipes

Commit:
Merge branch 'master' into refactor_show

Commit:
Merge pull request #60 from omidzargham/refactor_show

Refactor show

Commit:
Merge branch 'master' of https://github.com/omidzargham/Grocer-Plus into forgot_password

Commit:
Fixed test error

Commit:
Error checking where API does not give back recipes based on data provided

Commit:
Merge pull request #62 from omidzargham/forgot_password

Forgot password

Commit:
Reduce cog complexity of show

Commit:
Removed duplication in grocery_list and favorited_recipes

Commit:
Merge pull request #63 from omidzargham/refactor_show_more

Reduce cog complexity of show

Commit:
Update README.md

Commit:
Merge pull request #64 from omidzargham/update-readme

Update README.md

adnanhemani commented:
Add into your ignore file for simplecov to ignore config/initializers - I think that's dragging your code coverage down by quite a bit

adnanhemani commented:
Please add a test for Grocery#to_s (from the model) -  should be a trivial test case to make

adnanhemani commented:
You need to add more precise instructions on how to set up the application. I'm guessing you need to export the API keys before running "rails server" - and you forgot to add in instructions on how to run the migrations.

adnanhemani commented:
After adding those in, I think this should be good to go.

ChloeGuyue committed:
to_s test

Commit:
Update README.md

ahz0001 committed:
Update README.md

Improved description to include more detailed instructions for setting up the server.

ahz0001 committed:
Merge branch 'master' into fix-readme

Commit:
fixed home routing

ChloeGuyue committed:
Merge pull request #66 from omidzargham/test_to_s

to_s test

ahz0001 committed:
Merge pull request #65 from omidzargham/fix-readme

Update README.md

Commit:
Delete all groceries associated when deleting a recipe

Commit:
removed superfluous view code

nkompella committed:
Merge pull request #67 from omidzargham/fix-home-routing

fixed home routing

Commit:
Merge pull request #68 from omidzargham/fix_links_groceries

Delete all groceries associated when deleting a recipe

Commit:
rspec

Commit:
Tests

Commit:
fix link regression bug

Commit:
Merge pull request #69 from omidzargham/update_test

Update test

adnanhemani commented:
LGTM
